### ISSUE TITLE
Whitelist GMA files for FastDL download

### ISSUE THREAD

**Issue by Down-s** [Facepunch/garrysmod-requests#1817]
### Details
It would be great if we could allow players to download GMA files through, this would be useful for downloading only one file and mounting the GMA instead of downloading a large number of files.

**Comment by viral32111** [comment]
Why not just upload the GMA to the workshop and use resource.AddWorkshop() instead?

**Comment by VaasKahnGrim** [comment]
I agree, this would be really useful for those who have content thats not allowed on workshop but still want things to be nice and organized for downloads

**Comment by thegrb93** [comment]
I'm guessing this is to work-around workshop being slow/offline? Not sure this is the best solution.

**Comment by VaasKahnGrim** [comment]
its already been done sorta, the only differances is this wouldn't require people renaming the file to make it work

**Comment by Down-s** [comment]
> I'm guessing this is to work-around workshop being slow/offline? Not sure this is the best solution.

No, it's because I don't want to have a workshop content pack, I would rather have it all managed on my own web server. Using the workshop is a pain.

**Comment by WilliamVenner** [comment]
so i have to download the same addon twice over to join your server?

**Comment by VaasKahnGrim** [comment]
> so i have to download the same addon twice over to join your server?

I think servers building their own content packs like this are more likely going to be alot smarter about the content going on their server, unlike <insert random 12yo server owner with 300 addons from workshop and 100 addons from gmodstore>

**Comment by WilliamVenner** [comment]
if you regularly play gmod then chances are you already have those 300 addons installed and no one is going to play your server if they have to redownload a load of bloat they already went through the horrific pain of downloading once before

**Comment by Down-s** [comment]
> so i have to download the same addon twice over to join your server?

What? I'm talking about custom content.

**Comment by WilliamVenner** [comment]
you can't trust the community to just use this for custom content

**Comment by Down-s** [comment]
People do this anyway on the workshop, I don't see how this makes it more of a problem.

**Comment by VaasKahnGrim** [comment]
tbh if they tried using it for something already on workshop it only hurts their own server rep and load times. Besides FastDL is only really used for custom stuff anyways(assuming you don't have the server being ran by children)

Might aswell have a way to package the files into a single(or handful of packs) rather than downloading each individule file seperately

**Comment by wackod** [comment]
yes,   workshop is not perfect   You can't guarantee that the download speed is very fast in any region and at any time, but your own fastdl server, you can control it

**Comment by aStonePenguin** [comment]
Additionally, save the file as a checksum of the download URL and it solves #3364 & associated other FastDL issues.

Many people already do this. Having to decompress and cache the gma with lua sucks.
